File: DP/python/11053.py
n = int(input())
nums = list(map(int,input().split()))
dp = [1 for _ in range(n)]

for i in range(n-1):
    for j in range(i+1,n):
        if nums[j] > nums[i]: dp[j] = max(dp[j],dp[i]+1)
print(max(dp))

File: DP/javascript/11053.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
const nums = input[1].split(' ').map(Number);
const dp = new Array(n).fill(1);

for (let i = 1 ; i < n ; i++){
    for (let j = 0 ; j<i ; j++){
        if (nums[j]<nums[i]) dp[i] = Math.max(dp[i],dp[j]+1);
    }
}
console.log(Math.max(...dp));